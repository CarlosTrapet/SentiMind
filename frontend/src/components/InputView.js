import React from 'react';
import SubHeader from './SubHeader';
import InputBox from './InputBox';
import Button from './Button';

const InputView = props => {
  return (
    <div>
      <SubHeader subHeaderText="Type in a topic, find out what the twitterverse thinks" />
      <InputBox handleQueryInput={props.handleQueryInput} />
      <Button buttonText="Go!" handleSubmit={props.handleSubmit} />
    </div>
  );
};

export default InputView;
