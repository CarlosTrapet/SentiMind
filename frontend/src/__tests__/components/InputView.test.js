import React from 'react';
import { shallow } from 'enzyme';
import InputView from '../../components/InputView';

describe('InputView', () => {
  const mockHandleQueryInput = jest.fn();
  const mockHandleSubmit = jest.fn();
  const props = {
    handleQueryInput: mockHandleQueryInput,
    handleSubmit: mockHandleSubmit
  };
  const inputView = shallow(<InputView {...props} />);

  it('renders correctly', () => {
    expect(inputView).toMatchSnapshot();
  });

  it('renders a SubHeader component', () => {
    expect(inputView.find('SubHeader').exists()).toBe(true);
  });

  it('renders a InputBox component', () => {
    expect(inputView.find('InputBox').exists()).toBe(true);
  });

  it('renders a Button component', () => {
    expect(inputView.find('Button').exists()).toBe(true);
  });

  describe('passing props', () => {
    describe('InputBox props', () => {
      it('passes handleQueryInput down to InputBox', () => {
        expect(inputView.find('InputBox').prop('handleQueryInput')).toBe(
          mockHandleQueryInput
        );
      });

      describe('SubHeader props', () => {
        it('passes flexible SubHeaderText down to SubHeader', () => {
          expect(inputView.find('SubHeader').prop('subHeaderText')).toBe(
            'Type in a topic, find out what the twitter-verse thinks'
          );
        });
      });

      describe('Button props', () => {
        it('passes handleSubmit down to Button', () => {
          expect(inputView.find('Button').prop('handleSubmit')).toBe(
            mockHandleSubmit
          );
        });

        it('passes buttonText down', () => {
          expect(inputView.find('Button').prop('buttonText')).toEqual('Go!');
        });
      });
    });
  });
});
