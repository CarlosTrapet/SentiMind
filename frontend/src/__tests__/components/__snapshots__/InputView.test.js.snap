// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputView renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputView
    handleQueryInput={[Function]}
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SubHeader
          subHeaderText="Type in a topic, find out what the twitter-verse thinks"
/>,
        <InputBox
          handleQueryInput={[Function]}
/>,
        <br />,
        <TwitterTweetEmbed
          tweetId="983343562736717825"
/>,
        <Button
          buttonText="Click to find out!"
          handleSubmit={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "subHeaderText": "Type in a topic, find out what the twitter-verse thinks",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleQueryInput": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "tweetId": "983343562736717825",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonText": "Click to find out!",
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SubHeader
            subHeaderText="Type in a topic, find out what the twitter-verse thinks"
/>,
          <InputBox
            handleQueryInput={[Function]}
/>,
          <br />,
          <TwitterTweetEmbed
            tweetId="983343562736717825"
/>,
          <Button
            buttonText="Click to find out!"
            handleSubmit={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "subHeaderText": "Type in a topic, find out what the twitter-verse thinks",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleQueryInput": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "tweetId": "983343562736717825",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "Click to find out!",
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
